### ISSUE TITLE
Using AWS provider without alias throws Invalid reference error

### ISSUE THREAD

**Issue by PW999** [terraform-linters/tflint-ruleset-aws#1002]
Using following `config.tf`

```
terraform {
  required_version = ">= 1.5.7"
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 6.0"
    }
  }
}


provider "aws" {
  region = "eu-west-1"
  assume_role {
    role_arn = "arn:aws:iam::1111111111:role/assume_admin"
  }
  default_tags {
    tags = {
      ManagedBy = "Terraform"
    }
  }
}

```

and following `main.tf`
```
resource "aws_iam_user" "bedrock" {
  provider = aws

  name = "test"
}
```

When I run tflint with following `.tflint.hcl`
```
tflint {
  required_version = ">= 0.50"
}

plugin "aws" {
  enabled = true
  version = "0.44.0"
  source  = "github.com/terraform-linters/tflint-ruleset-aws"
}

rule "aws_resource_missing_tags" {
  enabled = true
  tags = ["layer", "type"]
  exclude = []
}
```

I get the following error
`Failed to check ruleset; failed to check "aws_resource_missing_tags" rule: main.tf:2,14-17: Invalid reference; A reference to a resource type must be followed by at least one attribute access, specifying the resource name.`

When I change `main.tf`to
```
resource "aws_iam_user" "bedrock" {
  provider = aws.test

  name = "test"
}
```
It does work and returns `main.tf:1:1: Notice - The resource is missing the following tags: "layer", "type". (aws_resource_missing_tags)`

Version info
```
tflint -v
TFLint version 0.60.0
+ ruleset.aws (0.44.0)
+ ruleset.terraform (0.13.0-bundled)
tf -v
OpenTofu v1.10.7
on linux_amd64
+ provider registry.opentofu.org/hashicorp/aws v6.24.0
```

**Comment by bendrucker** [comment]
There's no need to explicitly reference a default provider instance this way, just remove it to work around this for now:

```tf
provider = aws
```

This syntax is [undocumented](https://developer.hashicorp.com/terraform/language/block/provider) but Terraform does accept it.
